src/mongo/db/pipeline/document_source_skip_test.cpp

```diff
@@ -65,5 +65,28 @@ TEST_F(DocumentSourceSkipTest, ShouldPropagatePauses) {
     ASSERT_TRUE(skip->getNext().isEOF());
 }
 
+TEST_F(DocumentSourceSkipTest, SkipsChainedTogetherShouldNotOverFlowWhenOptimizing) {
+    // $skip should not optimize if overflow
+    auto skipShort = DocumentSourceSkip::create(getExpCtx(), 1);
+    auto skipLong = DocumentSourceSkip::create(getExpCtx(), std::numeric_limits<long long>::max());
```

Comment:
Can you add a test to make sure that two $skips with `std::numeric_limits<long long>::max()` will not combine?